stopes/modules/bitext/indexing/populate_faiss_index.py

```diff
@@ -289,8 +289,10 @@ def add_embedding_to_index(
                 # but is quite slow, and when we aren't afraid of preemption we should skip it.
                 # Write partial index to the partial index file (and while this happens, keep is_partial_file_valid as False)
                 checkpoint_summary.is_partial_file_valid = False
+                if gpu:
+                    partial_cpu_idx = faiss.index_gpu_to_cpu(partial_idx)
                 faiss.write_index(
-                    partial_idx,
```

Comment:
That's not going to work if it's not on GPU as partial_cpu_index is undefined unless it goes through the if.

Reply:
Done